src/allotropy/parsers/agilent_gen5/plate_data.py

```diff
@@ -358,12 +507,157 @@ def _blank_data_cube(self, measures: list[float]) -> TDatacube:
-        hours, minutes, seconds = tuple(int(num) for num in hhmmss.split(":"))
-        return (3600 * hours) + (60 * minutes) + seconds
 
-    @abstractmethod
+@dataclass
+class PlateData:
+    file_paths: FilePaths
+    plate_number: PlateNumber
+    plate_type: PlateType
+    results: Results
+    curve_name: CurveName
+    kinetic_data: KineticData
+
+    @staticmethod
+    def create(lines_reader: LinesReader) -> PlateData:
+        file_paths = FilePaths.create(lines_reader)
+        plate_number = PlateNumber.create(lines_reader)
+        plate_type = PlateType.create(lines_reader)
+        layout_data = LayoutData.create_default()
+        actual_temperature = ActualTemperature.create_default()
+        results = Results.create()
+        curve_name = CurveName.create_default()
+        kinetic_data = KineticData.create_default()
+
+        while lines_reader.current_line_exists():
+            data_section = read_data_section(lines_reader)
+            if data_section.startswith("Layout"):
+                layout_data = LayoutData.create(data_section)
+            elif data_section.startswith("Actual Temperature"):
+                actual_temperature = ActualTemperature.create(data_section)
+            elif data_section.startswith("Results"):
+                results.parse_results(
+                    data_section,
+                    plate_type,
+                    plate_number,
+                    layout_data,
+                    actual_temperature,
+                )
+            elif data_section.startswith("Curve Name"):
+                curve_name = CurveName.create(
+                    data_section,
+                    plate_number,
+                    results,
+                )
+            elif len(data_section.split("\n")) == 1 and any(
+                read_name in data_section for read_name in plate_type.read_names
+            ):
+                if data_section.startswith("Blank"):
+                    kinetic_data.parse_blank_kinetic_data(
+                        lines_reader,
+                        data_section.strip(),
+                        results,
+                        plate_type,
+                    )
+                else:
+                    kinetic_data = KineticData.create(
+                        lines_reader,
+                        results,
+                    )
+
+        return plate_type.experiment_cls(
+            file_paths=file_paths,
+            plate_number=plate_number,
+            plate_type=plate_type,
+            results=results,
+            curve_name=curve_name,
+            kinetic_data=kinetic_data,
+        )
+
+    @staticmethod
+    def get_read_mode() -> ReadMode:
+        raise NotImplementedError
+
+    @staticmethod
+    def get_data_point_cls() -> type[DataPoint]:
+        raise NotImplementedError
+
     def to_allotrope(self, measurement_docs: list) -> Any:
         raise NotImplementedError
+
+
+class AbsorbancePlateData(PlateData):
```

Comment:
Can we drop these classes, and instead make build the model inside the parser? Instead of returning a class corresponding to read mode, just return the PlateData, and store the plate_type.read_mode to decide which model to build.


This has a couple purposes: 

1. It more closely follows the structure of the other parsers (structure classes hold data, parser builds the models)
2. It paves the way for the new plate reader schema, where all of these will use the same model.